Assets/Scripts/IDeathListener.cs

```diff
@@ -0,0 +1,6 @@
+using UnityEngine;
+
+public interface IDeathListener
+{
+    public void OnDeath();
```

Comment:
Interfaces shall not declare visibility modifiers. Support for them was introduced in C# 8 and declares that you wish to add a default implementation (which you've omitted)

```cs
public interface IDeathListener
{
    // no default implementation
    void OnDeath();

    // with default implementation
    public void OnDeath()
    {
        Debug.Log("Died");
    }

    // something weird in-between
    public void OnDeath();
}